File: backend/src/main/java/com/izzydrive/backend/model/Message.java
package com.izzydrive.backend.model;

import com.izzydrive.backend.model.users.User;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Date;

@Entity
@Table(name="messages")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 2048)
    private String tring;

    @Column(nullable = false)
    private LocalDateTime timestamp;

    @ManyToOne(fetch = FetchType.LAZY)
    private User sender;

    @ManyToOne(fetch = FetchType.LAZY)
    private User recipient;

    @Column(name = "channel")
    private String channel;

    @Column(name = "read_date")
    private LocalDateTime readDate;
}

File: frontend/src/app/model/message/message.ts
export class Message{

    sender: string;
    text: string;
    timeStamp: string;
    channel?: string;
    channel_key?: string;

    constructor(sender: string,
        text: string,
        timeStamp: string,
        channel_key?: string,
        channel?: string){
        
        this.sender = sender;
        this.text = text;
        this.timeStamp = timeStamp;
        this.channel_key = channel_key;
        this.channel = channel
        }
}